src/compile.ts

```diff
@@ -219,7 +227,11 @@ export function compile(
           ]);
         } else if (ts.isIdentifier(node)) {
           if (node.text === "undefined" || node.text === "null") {
-            return newExpr("NullLiteralExpr", []);
+            return newExpr("NullLiteralExpr", [
+              ts.factory.createIdentifier(
+                node.text === "undefined" ? "true" : "false"
```

Comment:
I think you mean:
```ts
ts.factory.createTrue()
ts.factory.createFalse()
```

Interesting idea here - use `true` to identify `undefined` and `false` to identify `null`?  Should we instead just add an `UndefinedLiteral`?

Reply:
We can, I didn't want to refactor all of the code, but I like this. will cut a PR.

Reply:
#52